censys_validation.py

```diff
@@ -0,0 +1,126 @@
+# File: censys_validation.py
+#
+# Copyright (c) 2016-2023 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+import ipaddress
+
+import phantom.app as phantom
+
+from censys_consts import CENSYS_ERR_MSG_UNAVAILABLE, CENSYS_INT_ERR_MSG
+
+
+def parse_http_error(self, action_result, r):
+    if "json" not in r.headers.get("Content-Type", ""):
+        return ""
+
+    try:
+        resp_json = r.json()
+    except Exception:
+        return (
+            action_result.set_status(
+                phantom.APP_ERROR, "Unable to parse response as JSON"
+            ),
+            None,
+        )
+
+    message = "Server returned error with status: {}, Type: {}, Detail: {}".format(
+        resp_json.get("status", "NA"),
+        resp_json.get("error_type", "NA"),
+        resp_json.get("error", "NA"),
+    )
+
+    return action_result.set_status(phantom.APP_ERROR, message)
+
+
+def get_error_message_from_exception(self, e):
+    """
+    Get appropriate error message from the exception.
+    :param e: Exception object
+    :return: error message
+    """
+
+    error_code = None
+    error_msg = CENSYS_ERR_MSG_UNAVAILABLE
+
+    try:
+        if hasattr(e, "args"):
+            if len(e.args) > 1:
+                error_code = e.args[0]
+                error_msg = e.args[1]
+            elif len(e.args) == 1:
+                error_msg = e.args[0]
+    except Exception:
+        self.debug_print("Error occurred while fetching exception information")
+
+    if not error_code:
+        error_text = f"Error Message: {error_msg}"
+    else:
+        error_text = f"Error Code: {error_code}. Error Message: {error_msg}"
+
+    return error_text
+
+
+def validate_integer(action_result, parameter, key, allow_zero=False):
```

Comment:
Looks like the last argument is always `False` so this part of logic can be simplified. I would also move only `param = float(param) to the try/except and cover the rest with conditionals